src/solvers/quasi_Newton.jl

```diff
@@ -49,6 +49,7 @@ mutable struct QuasiNewtonState{
     retraction_method::RTR
     stepsize::S
     stop::SC
+    p_old::P
```

Comment:
Hm, is it too much effort to use a storage here? In (nearly) all other places where we need the last iterate, we use a storage.

Reply:
No, not too much effort, I just wanted to keep the code simple (and with storage, it's updating is still not intuitive for me). I can change it to storage. Anyway, more storage is needed but I only wanted to make a quick small improvement here.

Reply:
I have not yet checked how much that affects everything, but the tests do something different now since WolfeLineSearch now has one case that is not covered.

Not that storing `p_old` here would store the last iterate double (once in the storage once here) even if we share a storage among all things.

Sure the updatey thing is a bit difficult and should be simplified to just be called at the end of the step in `solve` I think (though I have to think about this a bit more).

I only feel that we spent so much time on storage that on these last few steps omitting that feels a bit - strange?

Reply:
Sadly, after all this work, I still don't see any real advantage of storage over a plain variable. Only centralized storage and updating would bring an advantage but it's still a lot of work to have it.

Plus, it's unclear to me if we even can have a single point of updating.

Reply:
Sure the updating is still a thing, that might not be so well-modelled and might still need work. But if we use storage in all places (even if for now those are separate copies) the step to centralise it is easier / shorter than to look though all code again whether we store some p_old – and sure here it was temporary in the code so having that not in there but in the state is beneficial.

Overall the goal is of course to just store p_old once (for debug/stop/step/stepsize...) instead of multiple copies.
The alternative is of course to have a lot of memory and allocations. But then we can omit this PR? Not my plan.

Reply:
Hm, I've tried converting that to storage and I can't make it look nice. I will just revert back to allocating code I think.

Reply:
Sure, maybe we should move this to the point where we have optimised the update (I have a few ideas but have to think about that as well) – something with a storage that keeps track of its “users”, they report that they are “ready to update” and if the last one reports that, the storage updates.

Reply:
In the meantime it would be nice to have "just update this thing" that looks nicer than
```julia
copyto!(M, get_storage(qns.storage, PointStorageKey(:p_old)), get_iterate(qns))
```
(and I'm not even sure if that's intended interface or not, but that's my best guess)

Reply:
currently there is `update_storage!(a,s)` ( see https://github.com/JuliaManifolds/Manopt.jl/blob/abedeeda37adc58a35e667bdbf5af24ccb413cca/src/plans/solver_state.jl#L264 ) but that of course updates the whole storage in the `AbstractStateAction a` (read as: Debug/Step/Stop/Record...) to the values given in the `State s`.
So even my first idea of putting a storage in a state `s´ does not directly work (as I proposed above).

But sure this could be reduced to have a `set_storage!(a, key, value)` that updates a key in the storage to value. With your pinky and vector key types this could even be used for those.
But you are right for now this does not exist, since I designed it to just “update all at the end.”
On the other hand no one than me used the code before, maybe not even looked at it, so it is definetly not an optimal design, for sure.

Reply:
Yes, `set_storage!(a, key, value)` would be very reasonable but let's leave it for another PR. I won't put more changes here.

Reply:
yes, I think that should be combined with making the storage (of different places) more interoperable as well and that's more for another PR.